src/distributions/mv_normal_mean_covariance.jl

```diff
@@ -88,7 +95,9 @@ function Base.prod(::ProdAnalytical, left::MvNormalMeanCovariance, right::MvNorm
     return MvNormalWeightedMeanPrecision(xi_left + xi_right, W_left + W_right)
 end
 
-function Base.prod(::ProdAnalytical, left::MvNormalMeanCovariance{T1}, right::MvNormalMeanCovariance{T2}) where {T1 <: LinearAlgebra.BlasFloat, T2 <: LinearAlgebra.BlasFloat}
+function Base.prod(
+    ::ProdAnalytical, left::MvNormalMeanCovariance{T1, <:AbstractVector, <:Matrix}, right::MvNormalMeanCovariance{T2, <:Vector, <:Matrix}
```

Comment:
Here is `AbstractVector`, but in the `MvNormalMeanPrecision` is `Vector`. Perhaps here should be `Vector` as well?

Reply:
You are absolutely right here. Should all be `AbstractVector`. I am fixing it right now.